### PR TITLE
Deprecate RAC2 support

### LINKED ISSUE #298: Deprecate RAC2 support?
Not insignificant time is being spent maintaining the RACSignal extensions, as well as the non-RAC3+ provider. We should consider officially marking these as deprecated for some time, before eventually removing them. 

Not sure if anyone else is using them or if its just me, and I won't be for much longer. Open for discussion!

### PR DESCRIPTION
**Pull request by petester42**
As the title states, ReactiveCocoa 2 support has been deprecated.

Closes #298

### CODE REVIEW

File: Source/ReactiveCore/MoyaError.swift [hunk 1]
```diff
@@ -9,9 +9,10 @@ public enum MoyaError: ErrorType {
     case Underlying(ErrorType)
 }
 
-// If RACSignal support gets removed we can also remove everything under here
+@available(*, deprecated, message="This will be removed when ReactiveCocoa 4 becomes final. Please visit https://github.com/Moya/Moya/issues/298 for more information.")
 public let MoyaErrorDomain = "Moya"
 
+@available(*, deprecated, message="This will be removed when ReactiveCocoa 4 becomes final. Please visit https://github.com/Moya/Moya/issues/298 for more information.")
 public enum MoyaErrorCode: Int {
```
Comment (swizzlr) [inline Source/ReactiveCore/MoyaError.swift]: I should think that, in fact, we can merely rename these types, remove the domain and take away the underlying `Int` – these types are still useful and ought to remain.

Comment (petester42) [inline Source/ReactiveCore/MoyaError.swift]: If the Int is taken away then it is just what MoyaError is. I don't really see a use for both MoyaErrorCode and MoyaError when we remove RAC 2 support. The only use of MoyaErrorCode is to generate an NSError for RAC 2.

**Comment by orta** [comment]
Probably worth mentioning in the README too

**Comment by orta** [comment]
Form maybe clarifying that its 3 and above, rather than saying what isn't supported

File: Source/ReactiveCore/MoyaError.swift (hunk 1 above)
Comment (ashfurrow) [inline Source/ReactiveCore/MoyaError.swift]: Something like https://github.com/Moya/Moya/blob/master/Source/ReactiveCore/MoyaError.swift ?

**Comment by ashfurrow** [comment]
Agreed with Orta – some clarification in the readme would be good, but that's not a blocker. I'll open an issue.

**Pull request merged**